Removed sphinx and enum34 requirements

sachit-wp opened the pull request.

Commit:
Removed sphinx and enum34 requirements

stonier commented:
Nice that the enum34 was a simple dependency fix. 

Only need to check around the requirements.txt files for which I'll have to play around and dig back through my memory. The `rtd` in `rtd-requirements.txt` was for `read-the-docs` (generates the py_trees documentation online). If I recall, it was necessary for read-the-docs, but in general, the `setup.py` requires should be sufficient. 

Which doesn't explain why I was calling on the rtd-requirements in a virtualenv script...

stonier commented:
Conondrums

* [x] Confirm `pip install -r rtd-requirements.txt` should be unnecessary for the virtualenv bash scripts - yes: `python setup.py develop` ensures you get dependencies in `install_requires`
* [x]  How do the virtualenv scripts get extra_requires dependencies? They can't. `python setup.py test` will get all the dependencies, but it drops them into your projects `.egg` folder instead.

stonier committed:
[infra] make sure rtd and virtualenv are completely sated wrt dependencies

This uses the rtd-requirements.txt as the goto for making sure
you have all your dependencies. What is in setup.py with install_requires
tests_require and extra_requires form the blueprint for what is
correctly made available from pypi and for typical setuptools (non-virtualenv)
style development.

And yes, python has too many ways to do the same thing.

sachit-wp commented:
To get the extra dependencies currently the only way to do so is as follows:
cd into the local directory
pip install -e .[put-the-name-here]

in this case we would have to do something like:
pip install -e .[test]
pip install -e .[docs]

If need be, we could probably merge them into one group - develop and then just do something like this:
pip install -e .[develop]

stonier committed:
[infra] use setup.py as the virtualenv source of truth for deps

This drops a use case for the rtd-requirements.txt file (Id like
to get rid of it eventually if at all possible, but this depends
on how the read-the-docs server needs things).

stonier commented:
Yeah, just discovered that myself. Updated the virtualenv scripts to call it like that so the rtd-requirements.txt can serve solely as the read-the-docs server provider. Would like to get rid of that eventually if possible as well.

stonier commented on README.md:
```diff
@@ -77,3 +77,10 @@ or in a catkin environment alongside other ROS py-trees packages:
 
 * https://github.com/stonier/repos_index/blob/devel/kinetic/py_trees.repos
 
+#### Optional
```
I'd drop this section. We provide virtualenvs that enable testing/doc generation and I typically prefer to avoid instructions that easily lead to people mucking up their local environments.

stonier commented on setup.py:
```diff
@@ -5,17 +5,22 @@
+    'enum34;python_version<"3.4"',
     'pydot'
 ]
+extras_require = {} if os.environ.get('CATKIN_BINARY_DIR') else {
```
I'd throw in a tests_require here as well and then re-use it in the extras_require and the tests_require further down below.

Commit:
Addressed changes as requested

stonier committed:
[infra] bump tests_require to reuse it

stonier reviewed (approved):
Bumped the tests_require ahead of the extras_require so it can be reused.

Good to go!
